privacyidea/lib/tokens/smstoken.py

```diff
@@ -338,9 +344,10 @@ def _send_sms(self, message="<otp>"):
         :return: submitted message
         :rtype: string
         """
-        ret = None
-
-        phone = self.get_tokeninfo("phone")
+        if is_true(self.get_tokeninfo("dynamic_phone")):
+            phone = self.user.get_user_phone("mobile")
+        else:
+            phone = self.get_tokeninfo("phone")
```

Comment:
Likewise, should we check for an empty phone number?

Reply:
I just pushed it...